test/src/audits/performance/user-timing.js

```diff
@@ -18,23 +18,17 @@
 const Audit = require('../../../../src/audits/performance/user-timings.js');
 const assert = require('assert');
 const traceContents = require('./trace-user-timings.json');
-
 /* eslint-env mocha */
+
 describe('Performance: user-timings audit', () => {
   it('fails gracefully', () => {
     const output = Audit.audit({});
     assert.equal(output.value, -1);
   });
 
-  it('processes a trace file for user timing data', () => {
+  it('evaluates valid input correctly', () => {
     const output = Audit.audit({traceContents});
-    assert.equal(output.value, 1);
-    assert.equal(output.extendedInfo.value[0].name, 'elapsed');
-  });
-
-  it('handles valid traces with no user timing data', () => {
-    const output = Audit.audit({traceContents: traceContents.slice(0, 1)});
-    assert.equal(output.value, 0);
-    assert.equal(output.extendedInfo.value.length, 0);
+    console.log(output);
+    assert.equal(output.value, 2);
```

Comment:
you can add the Closure record type discussed in https://github.com/GoogleChrome/lighthouse/pull/407/files#r65617183 as `UserTimingExtendedInfo` or whatever and pull the `mainThreadEvents` loop out into its own function that returns `UserTimingExtendedInfo` so you'll get type checking on the `extendedInfo.value` and you only need to check that the numbers you get are correct, not the exact object format